src/taraxad/main.cpp

```diff
@@ -10,60 +14,533 @@ using namespace std;
 
 namespace bpo = boost::program_options;
 
-int main(int argc, const char* argv[]) {
+string getHomeDir() {
+  struct passwd *pw = getpwuid(getuid());
+  const char *homedir = pw->pw_dir;
+
+  return string(homedir);
+}
+
+string getTaraxaDefaultDir() {
+  string home_dir = getHomeDir();
+  auto taraxa_dir = home_dir + "/.taraxa";
+  fs::create_directory(taraxa_dir);
+
+  return taraxa_dir;
+}
+
+string getAccountFile() { return getTaraxaDefaultDir() + "/account.json"; }
+
+string getVrfFile() { return getTaraxaDefaultDir() + "/vrf.json"; }
+
+bool fileExist(string file) {
+  ifstream ifile;
+  ifile.open(file);
+  if (ifile) {
+    ifile.close();
+    return true;
+  } else {
+    return false;
+  }
+}
+
+bool writeFile(string file_name, Json::Value json) {
+  ofstream ofile(file_name);
+  if (ofile.is_open()) {
+    ofile << json;
+    return true;
+  } else {
+    cerr << "Cannot open file " << file_name << endl;
+    return false;
+  }
+}
+
+Json::Value getAccountJson(KeyPair account) {
+  Json::Value json(Json::objectValue);
+  json["node_secret"] = toHex(account.secret().ref());
+  json["node_public"] = account.pub().toString();
+  json["node_address"] = account.address().toString();
+
+  return json;
+}
+
+Json::Value getVrfJson(taraxa::vrf_wrapper::vrf_sk_t sk, taraxa::vrf_wrapper::vrf_pk_t pk) {
+  Json::Value json(Json::objectValue);
+  json["vrf_secret"] = sk.toString();
+  json["vrf_public"] = pk.toString();
+
+  return json;
+}
+
+Json::Value getAccountSecretKey() {
+  auto account_file = getAccountFile();
+  ifstream ifile(account_file);
+  if (ifile.is_open()) {
+    Json::Value account_json;
+    ifile >> account_json;
+    ifile.close();
+    return account_json["node_secret"];
+  } else {
+    cerr << "Cannot open file " << account_file << endl;
+    exit(1);
+  }
+}
+
+Json::Value getVrfSecretKey() {
+  auto vrf_file = getVrfFile();
+  ifstream ifile(vrf_file);
+  if (ifile.is_open()) {
+    Json::Value vrf_json;
+    ifile >> vrf_json;
+    ifile.close();
+    return vrf_json["vrf_secret"];
+  } else {
+    cerr << "Cannot open file " << vrf_file << endl;
+    exit(1);
+  }
+}
+
+Json::Value getNetworkBootNodes(int network_identifier) {
+  Json::Value network_boot_nodes(Json::arrayValue);
+
+  switch (network_identifier) {
+    case 1: {
+      cout << "Mainnet has not supported yet" << endl;
+      exit(1);
+    }
+    case 2: {
+      Json::Value boot_node1(Json::objectValue);
+      boot_node1["id"] =
+          "1105725ff38294f42197a3851c87631610956404678a25f90da5f989bb590413a62665cadde1b820de3e9098367909110bba6d62d248"
+          "444ce45547a703bd8d10";
+      boot_node1["ip"] = "boot-node-0.testnet.taraxa.io";
+      boot_node1["tcp_port"] = 10002;
+      boot_node1["udp_port"] = 10002;
+      network_boot_nodes.append(boot_node1);
+      Json::Value boot_node2(Json::objectValue);
```

Comment:
how about make this loop with configurable num of nodes ? it will be smaller and easier to change in case we want to try different set of nodes